### PR TITLE
Update/refactor docs

### PR DESCRIPTION
**Pull request by ramineni**
This commit is to update go version and also refactor some parts
of doc

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: docs/getting-started-provider-dev.md [hunk 1]
```diff
@@ -145,10 +145,22 @@ and versioning information.
...
 
-Following the [GitHub Workflow](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md)
-guidelines for Kubernetes development, set up your environment and get the latest development repository. Begin
-by forking both the Kubernetes and Cloud-Provider-OpenStack projects into your GitHub into your local
-workspace (or bringing your current fork up to date with the current state of both repositories).
+Following the [GitHub Workflow](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md) guidelines for Kubernetes development, set up your environment and get the latest development repository. Begin by forking both the Kubernetes and Cloud-Provider-OpenStack projects into your GitHub into your local workspace (or bringing your current fork up to date with the current state of both repositories).
+
+`make` will build, test, and package this project. This project uses [go dep](https://golang.github.io/dep/) for dependency management.
+
+If you don't have a Go Environment setup, we also offer the ability to run make
```
Comment (adisky) [inline docs/getting-started-provider-dev.md]: could we add this part as a separate heading?? it is little deviating from the flow.

**Comment by adisky** [comment]
@ramineni Thanks, mostly looks good to me, just one suggestion inline

File: docs/getting-started-provider-dev.md (hunk 1 above)
Comment (ramineni) [reply docs/getting-started-provider-dev.md]: ok

**Comment by ramineni** [comment]
@adisky updated. PTAL

**Comment by adisky** [comment]
> @adisky updated. PTAL

@ramineni Thanks!!

**Pull request merged**